modules/magma/CreateInfo.hpp

```diff
@@ -80,6 +80,41 @@ namespace magma
     }
   };
 
+  ///
+  /// \brief Constructs a CreateInfo struct using the given arguments
+  ///
+  /// If present, `stype` and `pnext` are auto-filled by vulkan-hpp, and should not be given
+  /// Lists are automaticly decomposed into a (size, pointer) pair
+  /// Supported list types are:
+  /// - std::array
+  /// - std::vector
+  /// - magma::SingleValListRef (should be created using asListRef)
+  /// - magma::EmptyList (passes (0, nullptr))
+  ///
+  template<class CreateInfo, class... T>
+  constexpr CreateInfo makeCreateInfo(T &&... args)
```

Comment:
probably noexceptness if the StructBuilder<...>::make(...) is noexcept

Reply:
Also it's seem's to be a factory.
make_unique and make_shared are also factory, why not using make_shared or make_unique in this case.

Also if you dont want to store a pointer you can just call regular constructor.

Reply:
I don't think I got what you mean. The struct is never instanciated, it serves only to store templated data. The `make` function of tha structure will then convert certain specific argument to pairs of size, data (not `std::pair`, the amount of parameters grow) and call the constructor of a Vulkan-hpp structure with it. I don't see how that is achievable with `std::make_{unique,shared}`. If it is achievable but the `StructBuilder<>` structure must be exposed I would be against changing.